### PR TITLE
[WIP] Variables for texts and footers

### PR DESCRIPTION
**Pull request by teolemon**
Adding constant in the config file to allow to generate and update static content, footers and more. An alternative approach would be to store in a taxonomy.
- [ ]  Logo urls
- [ ]  Key figures that change overtime (but that we don't want to update dynamically)

and possibly more.

### CODE REVIEW

File: lib/ProductOpener/Config_off.pm [hunk 1]
```diff
@@ -393,6 +393,173 @@ my @related_applications = (
 	{ 'platform' => 'windows', 'id' => '9nblggh0dkqr', 'url' => 'https://www.microsoft.com/p/openfoodfacts/9nblggh0dkqr' },
 );
 
+my @localized_stores_icons = (
+	{ 'platform' => 'windows', 'ar', 'image_name' => 'Arabic.svg' },
+	{ 'platform' => 'windows', '', 'image_name' => 'Belarusian.svg' },
```
Comment (teolemon) [inline lib/ProductOpener/Config_off.pm]:
```suggestion
	{ 'platform' => 'windows', 'by', 'image_name' => 'Belarusian.svg' },
```

File: lib/ProductOpener/Config_off.pm [hunk 2]
```diff
@@ -393,6 +393,173 @@ my @related_applications = (
...
+	{ 'platform' => 'windows', '', 'image_name' => 'Bengali.svg' },
+	{ 'platform' => 'windows', '', 'image_name' => 'Bosnian.svg' },
+	{ 'platform' => 'windows', 'bg', 'image_name' => 'Bulgarian.svg' },
+	{ 'platform' => 'windows', 'zh', 'image_name' => 'Chinese_Simplified.svg' },
+	{ 'platform' => 'windows', '', 'image_name' => 'Croatian.svg' },
```
Comment (teolemon) [inline lib/ProductOpener/Config_off.pm]:
```suggestion
	{ 'platform' => 'windows', 'hr', 'image_name' => 'Croatian.svg' },
```

File: lib/ProductOpener/Config_off.pm [hunk 3]
```diff
@@ -393,6 +393,173 @@ my @related_applications = (
...
+	{ 'platform' => 'windows', '', 'image_name' => 'Danish.svg' },
+	{ 'platform' => 'windows', 'en', 'image_name' => 'English.svg' },
+	{ 'platform' => 'windows', '', 'image_name' => 'Estonian.svg' },
+	{ 'platform' => 'windows', '', 'image_name' => 'Filipino.svg' },
+	{ 'platform' => 'windows', 'fr', 'image_name' => 'French.svg' },
+	{ 'platform' => 'windows', 'el', 'image_name' => 'Greek.svg' },
+	{ 'platform' => 'windows', 'hi', 'image_name' => 'Hindi.svg' },
+	{ 'platform' => 'windows', '', 'image_name' => 'Indonesian.svg' },
```
Comment (teolemon) [inline lib/ProductOpener/Config_off.pm]:
```suggestion
	{ 'platform' => 'windows', 'id', 'image_name' => 'Indonesian.svg' },
```

File: lib/ProductOpener/Config_off.pm [hunk 4]
```diff
@@ -393,6 +393,173 @@ my @related_applications = (
...
+      { 'platform' => 'windows', 'ar', 'image_name' => 'Arabic.svg' },
+      { 'platform' => 'windows', 'be', 'image_name' => 'Belarusian.svg' },
+      { 'platform' => 'windows', '', 'image_name' => 'Bengali.svg' },
+      { 'platform' => 'windows', '', 'image_name' => 'Bosnian.svg' },
+      { 'platform' => 'windows', 'bg', 'image_name' => 'Bulgarian.svg' },
+      { 'platform' => 'windows', 'zh', 'image_name' => 'Chinese_Simplified.svg' },
+      { 'platform' => 'windows', 'hr', 'image_name' => 'Croatian.svg' },
+      { 'platform' => 'windows', '', 'image_name' => 'Danish.svg' },
+      { 'platform' => 'windows', 'en', 'image_name' => 'English.svg' },
+      { 'platform' => 'windows', '', 'image_name' => 'Estonian.svg' },
```
Comment (teolemon) [inline lib/ProductOpener/Config_off.pm]:
```suggestion
      { 'platform' => 'windows', 'ee', 'image_name' => 'Estonian.svg' },
```

**Comment by stephanegigandet** [comment]
I would prefer a more general approach using the internationalization we already have in place: we can generate a set of .po files for all the language specific strings (e.g. image file names, app-store urls etc.) and then add a way to include any translation (whether from the .po files from crowdin, or the new po files).

That way we don't put translations in config files, and we can extend this to any kind of variable, without having new code.

**Comment by teolemon** [comment]
This approach has failed for tags.pot
Those variable are unlikely to change. We could use them to generate po files if you like, but leaving it to translation will introduce potential for bugs, extra work…

**Comment by teolemon** [comment]
image
This is an example (out of many) of why I want to remove predictable stuff from the po files.

**Comment by teolemon** [comment]
Given no consensus, I'm closing this PR, and given the effort I've put into it, I'm putting it in a bug, so it can serve when we converge on a sane solution to i18n

**Pull request closed without merging**